homeassistant/components/generic/config_flow.py

```diff
@@ -0,0 +1,317 @@
+        container = await hass.async_add_executor_job(
+            partial(
+                av.open,
+                stream_source,
+                options=stream_options,
+                timeout=SOURCE_TIMEOUT,
+            )
+        )
+        _ = container.streams.video[0]
+    except (av.error.FileNotFoundError):  # pylint: disable=c-extension-no-member
+        return {CONF_STREAM_SOURCE: "stream_file_not_found"}
+    except (av.error.HTTPNotFoundError):  # pylint: disable=c-extension-no-member
+        return {CONF_STREAM_SOURCE: "stream_http_not_found"}
+    except (av.error.TimeoutError):  # pylint: disable=c-extension-no-member
+        return {CONF_STREAM_SOURCE: "timeout"}
+    except av.error.HTTPUnauthorizedError:  # pylint: disable=c-extension-no-member
+        return {CONF_STREAM_SOURCE: "stream_unauthorised"}
+    except (KeyError, IndexError):
+        return {CONF_STREAM_SOURCE: "stream_novideo"}
+    except PermissionError:
+        return {CONF_STREAM_SOURCE: "stream_not_permitted"}
+    except OSError as err:
+        if err.errno == EHOSTUNREACH:
+            return {CONF_STREAM_SOURCE: "stream_no_route_to_host"}
+        if err.errno == EIO:  # input/output error
+            return {CONF_STREAM_SOURCE: "stream_io_error"}
+        raise err
+    return {}
+
+
+class GenericIPCamConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Config flow for generic IP camera."""
+
+    VERSION = 1
+
+    @staticmethod
+    def async_get_options_flow(
+        config_entry: ConfigEntry,
+    ) -> GenericOptionsFlowHandler:
+        """Get the options flow for this handler."""
+        return GenericOptionsFlowHandler(config_entry)
+
+    def check_for_existing(self, options):
+        """Check whether an existing entry is using the same URLs."""
+        return any(
+            entry.options[CONF_STILL_IMAGE_URL] == options[CONF_STILL_IMAGE_URL]
+            and entry.options[CONF_STREAM_SOURCE] == options[CONF_STREAM_SOURCE]
+            for entry in self._async_current_entries()
+        )
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the start of the config flow."""
+        errors = {}
+        if user_input:
+            # Secondary validation because serialised vol can't seem to handle this complexity:
+            if not user_input.get(CONF_STILL_IMAGE_URL) and not user_input.get(
+                CONF_STREAM_SOURCE
+            ):
+                errors["base"] = "no_still_image_or_stream_url"
+            else:
+                errors, still_format = await async_test_still(self.hass, user_input)
+                errors = errors | await async_test_stream(self.hass, user_input)
+                still_url = user_input.get(CONF_STILL_IMAGE_URL)
+                stream_url = user_input.get(CONF_STREAM_SOURCE)
+                name = user_input.get(
+                    CONF_NAME, still_url or stream_url or DEFAULT_NAME
+                )
```

Comment:
Maybe drop the protocol with urlparse, and then slugify this?

Reply:
We could also make the name `entry.title` instead and setup a reload listener so when they change the title it changes the entity name

Reply:
I'm not sure I follow.  Slugify sans protocol makes sens - I'm looking into that. But the `name` value is being used to set the `entity.title` already.  Please could you clarify?

Reply:
Example

https://github.com/home-assistant/core/pull/52360/files#r825511934